File: jumps1.java
class Solution {
    public boolean canJump(int[] nums) {
        if(nums == null || nums.length == 0) return true;
        int n = nums.length - 1;
        int dest = n; 
        
        if(dest == 0) return true;    
        for(int i = n - 1; i >= 0; i--){
            if(nums[i] + i >= dest){
                dest = i;
            }
            if(dest == 0) return true;
        }
        if(dest != 0) return false;
        return true;
    }
}

File: jumps1.py
# // Time Complexity :O(n)
# // Space Complexity :O(1)
# // Did this code successfully run on Leetcode :yes
# // Any problem you faced while coding this :no


# // Your code here along with comments explaining your approach

class Solution:
    def canJump(self, nums: List[int]) -> bool:
        destination=len(nums)-1
        for i in range(len(nums)-2,-1,-1):
            print(destination)
            if i+nums[i]>=destination:
                print("x")
                destination=i
        return destination==0
        